arrow/src/test/java/org/apache/calcite/adapter/arrow/ArrowAdapterTest.java

```diff
@@ -373,7 +373,7 @@ static void initializeArrowState(@TempDir Path sharedTempDir)
     String sql = "select * from arrowdata\n"
         + " where \"floatField\"=15.0";
     String plan = "PLAN=ArrowToEnumerableConverter\n"
-        + "  ArrowFilter(condition=[=(CAST($2):DOUBLE, 15.0)])\n"
+        + "  ArrowFilter(condition=[=(CAST($2):DOUBLE, 15.0E0)])\n"
```

Comment:
I see a lot of `E0` added. Is it a goal of this change to make decimal literals print differently from double literals?

Maybe I missed it, but somewhere in the javadoc (e.g. for RexLiteral) should state that goal.

Reply:
I haven't changed how floating point values are serialized as text. The difference comes because now more literals contain a double/float.

Reply:
Makes sense.

+1 when you fix the issues in my other 2 comments